build-tools/packages/build-tools/src/type-test-generator/typeTestUtils.ts

```diff
@@ -0,0 +1,291 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { readJsonSync } from "fs-extra";
+import path from "node:path";
+import { IExtractorConfigPrepareOptions } from "@microsoft/api-extractor";
```

Comment:
Anything in here using api-extractor in here should be reconsidered. We use api-extractor as a means to an end, but that is not a requirement to deliver the pattern we are using. Customers won't be loading api-extractor to use FF packages; so, we shouldn't need to use it to check compatibility.

Reply:
We're using API-Extractor here as an _option_; it isn't a hard requirement. I.e., if a config is found, and there is configured output for it, we use it to find the appropriate typings rollup for validation.

Reply:
This pattern was recommended early on in the design by @tylerbutler. Things have evolved since then, but I think this pattern is probably fine for now. I would prefer we save any further changes for a future PR.

Reply:
Unfortunately, this approach doesn't work as it is. The previous versions of packages have the api-extractor files, but they all inherit from base configs using _relative paths_ which are broken for the previous versions that get installed in node_modules. When testing this PR, I see errors like this:

```
Error: Error loading /Volumes/Code/Fluid-bt/azure/packages/azure-service-utils/common/build/build-common/api-extractor-base.json:
File does not exist: /Volumes/Code/Fluid-bt/azure/packages/azure-service-utils/common/build/build-common/api-extractor-base.json
File does not exist: /Volumes/Code/Fluid-bt/azure/packages/azure-service-utils/common/build/build-common/api-extractor-base.json
ENOENT: no such file or directory, open '/Volumes/Code/Fluid-bt/azure/packages/azure-service-utils/common/build/build-common/api-extractor-base.json'
```

My memory is a bit fuzzy but as I recall we were using this to find the alpha types to load, but reading the exports is what we ultimately need to do. I agree with a "merge and iterate" approach but I think the PR is broken at the moment. @WayneFerrao FYI.

Reply:
We should not use api-extractor. I didn't think it was just a hint, but if it is just a hint, then it should be easy to remove.
I am hoping to remove roll-up files from client repo tomorrow per the issues we run into. So, we should just steer clear.

Reply:
See [export function getApiExports(sourceFile: SourceFile): ExportRecords](https://github.com/WayneFerrao/FluidFramework/blob/9c4268ffa2130b1adc93d433c776f2d7df728e26/build-tools/packages/build-cli/src/library/typescriptApi.ts#L91-L92) which is underlying helper we use in two other places within `flub` to work with imports. Should get a lot of the information you need. Can probably be extended if there is more that you need.